joblib/memory.py

```diff
@@ -829,14 +819,66 @@ def __init__(self, location=None, backend='local', cachedir=None,
                               "cachedir value will be ignored.",
                               DeprecationWarning, stacklevel=2)
 
-        if isinstance(location, _basestring):
-            location = os.path.join(location, 'joblib')
-
         self.store_backend = _store_backend_factory(
-            backend, location, verbose=self._verbose,
-            backend_options=dict(compress=compress, mmap_mode=mmap_mode,
+            backend, self._location(location), verbose=self._verbose,
+            backend_options=dict(compress=compress,
+                                 mmap_mode=mmap_mode,
                                  **backend_options))
 
+    # ------------------------------------------------------------------------
+    # Public interface
+    # ------------------------------------------------------------------------
+
+    def clear(self, warn=True):
+        """Erase the complete cache directory."""
+        if warn:
+            self.warn('Flushing completely the cache')
+        if self.store_backend is not None:
+            self.store_backend.clear()
+
+    # ------------------------------------------------------------------------
+    # Private `object` interface
+    # ------------------------------------------------------------------------
+
+    def __repr__(self):
+        return '{0}(location={1})'.format(
+            self.__class__.__name__, (repr(None) if self.store_backend is None
+                                      else repr(self.store_backend)))
+
+    def __reduce__(self):
+        """ We don't store the timestamp when pickling, to avoid the hash
+            depending from it.
+            In addition, when unpickling, we run the __init__
+        """
+        # We need to remove 'joblib' from the end of cachedir
+        location = (repr(self.store_backend)[:-7]
+                    if self.store_backend is not None else None)
+        compress = self.store_backend.compress \
+            if self.store_backend is not None else False
+        return (self.__class__, (location, self.backend, self.mmap_mode,
+                                 compress, self._verbose))
+
+    def _location(self, location):
+        if isinstance(location, _basestring):
+            return os.path.join(location, self._root)
+        return location
+
+
+###############################################################################
+# class `Memory`
+###############################################################################
+class Memory(StoreBase):
+    """ A context object for caching a function's return value each time it
+        is called with the same input arguments.
+
+        All values are cached on the filesystem, in a deep directory
+        structure.
+
+        see :ref:`memory_reference`
```

Comment:
Hum, did we loose the docstring?

Reply:
It was already like this IIRC. The `__init__` has moved to `StoreBase`